### ISSUE TITLE
service_browser failed: DNS failure: NXDOMAIN

### ISSUE THREAD

**Issue by devZer0** [lathiat/avahi#172]
from time to time (depending who's attached to the network) i cannot browse anymore and get 

# avahi-browse -a
service_browser failed: DNS failure: NXDOMAIN

also see:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=883276

**Comment by devZer0** [comment]
please provide a command-line option to disable dns lookups or don't quit if  name lookup fails

# avahi-browse -a
≈+ ens192 IPv4 KONICA MINOLTA bizhub C284e                   UNIX Printer         local
+ ens192 IPv4 KONICA MINOLTA bizhub C284e                   PDL Printer          local
+ ens192 IPv4 jenkins (2)                                   _hudson._tcp         local
+ ens192 IPv4 jenkins (2)                                   _jenkins._tcp        local
service_browser fehlgeschlagen: DNS-Fehler: NXDOMAIN

tcpdump:
 8:16:00.555714 IP 172.16.26.72.56288 > 172.16.30.10.domain: 20358+ PTR? _net-assistant._udp.1009672797.members.btmm.icloud.com. (72)
18:16:00.556620 IP 172.16.30.10.domain > 172.16.26.72.56288: 20358 NXDomain 0/1/0 (131)
18:16:00.562931 IP 172.16.26.72.56288 > 172.16.30.10.domain: 20359+ PTR? _rfb._tcp.1009672797.members.btmm.icloud.com. (62)
18:16:00.563588 IP 172.16.30.10.domain > 172.16.26.72.56288: 20359 NXDomain 0/1/0 (121)

**Comment by evverx** [comment]
> please provide a command-line option to disable dns lookups

Something like that was also requested in https://github.com/avahi/avahi/issues/561 to make it possible to pass `--proto=mdns` (or something like that) to `avahi-resolve` to force avahi to use `AVAHI_LOOKUP_USE_MULTICAST` and avoid the unicast DNS entirely. `avahi-browse` should probably be updated too.

> don't quit if name lookup fails - as you can see the domain lookup error makes avahi-browse unusable

It's a bug and it's kind of bad that it's possible to bring down avahi-browse by sending unsolicited PTR RRs pointing `_services._dns-sd._udp.local` to unresolvable names.

**Comment by pemensik** [comment]
Yes, I think we want to completely separate local mDNS world from wide-area. It should not be possible by default to mix those two together. That could lead only to strange issues and potential for security attacks.
